src/storage/mvcc/txn.rs

```diff
@@ -212,14 +250,7 @@ impl<S: Snapshot> MvccTxn<S> {
 
             // Check data constraint when acquiring pessimistic lock.
             if should_not_exist {
-                if write.write_type == WriteType::Put
-                    || (write.write_type != WriteType::Delete
-                        && self.key_exist(&key, write.start_ts - 1)?)
-                {
-                    return Err(Error::AlreadyExist {
-                        key: key.into_raw()?,
-                    });
-                }
+                self.check_data_constraint(&write, &key)?;
```

Comment:
I think it's better to put `should_not_exist` as an argument to `check_data_constraint`.

Reply:
fixed